src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpTagHelper.cs

```diff
@@ -0,0 +1,117 @@
+        /// </summary>
+        /// <param name="method"><see cref="HttpMethod"/>.</param>
+        /// <returns>Span operation name.</returns>
+        public static string GetOperationNameForHttpMethod(HttpMethod method) => HttpMethodOperationNameCache.GetOrAdd(method, ConvertHttpMethodToOperationNameRef);
+
+        /// <summary>
+        /// Gets the OpenTelemetry standard method name for a span based on its <see cref="HttpMethod"/>.
+        /// </summary>
+        /// <param name="method"><see cref="HttpMethod"/>.</param>
+        /// <returns>Span method name.</returns>
+        public static string GetNameForHttpMethod(HttpMethod method) => HttpMethodNameCache.GetOrAdd(method, ConvertHttpMethodToNameRef);
+
+        /// <summary>
+        /// Gets the OpenTelemetry standard version tag value for a span based on its protocol <see cref="Version"/>.
+        /// </summary>
+        /// <param name="protocolVersion"><see cref="Version"/>.</param>
+        /// <returns>Span flavor value.</returns>
+        public static string GetFlavorTagValueFromProtocolVersion(Version protocolVersion) => ProtocolVersionToStringCache.GetOrAdd(protocolVersion, ConvertConvertProtcolVersionToStringRef);
+
+        /// <summary>
+        /// Gets the OpenTelemetry standard status code tag value for a span based on its protocol <see cref="HttpStatusCode"/>.
+        /// </summary>
+        /// <param name="statusCode"><see cref="HttpStatusCode"/>.</param>
+        /// <returns>Span status code value.</returns>
+        public static string GetStatusCodeTagValueFromHttpStatusCode(HttpStatusCode statusCode) => StatusCodeToStringCache.GetOrAdd(statusCode, ConvertHttpStatusCodeToStringRef);
+
+        /// <summary>
+        /// Gets the OpenTelemetry standard host tag value for a span based on its request <see cref="Uri"/>.
+        /// </summary>
+        /// <param name="requestUri"><see cref="Uri"/>.</param>
+        /// <returns>Span host value.</returns>
+        public static string GetHostTagValueFromRequestUri(Uri requestUri)
+        {
+            string host = requestUri.Host;
+
+            if (requestUri.IsDefaultPort)
+            {
+                return host;
+            }
+
+            int port = requestUri.Port;
+
+            if (!HostAndPortToStringCache.TryGetValue(host, out ConcurrentDictionary<int, string> portCache))
+            {
+                portCache = new ConcurrentDictionary<int, string>();
+                HostAndPortToStringCache.TryAdd(host, portCache);
+            }
+
+            if (!portCache.TryGetValue(port, out string hostTagValue))
+            {
+                hostTagValue = $"{requestUri.Host}:{requestUri.Port}";
+                portCache.TryAdd(port, hostTagValue);
+            }
+
+            return hostTagValue;
+        }
+
+        private static string ConvertMethodToOperationName(string method) => $"HTTP {method}";
```

Comment:
I interpret the addition of "HTTP " as a suggestion of the spec, in this case, it seems reasonable to directly use the `Method` field and avoid all caching and related. Or at least have an option for it. We can have a better conversation about it in a separate PR.

Reply:
That was a more recent change. Check out: #633. I agree having options to control it would be great. It was even talked about on that PR. I just went in and tried to make that work, but there are a few ways to do it we should probably discuss first. All of this logic is static, so I would need static options. Not really a big deal, but that would only work for this particular ActivitySource which we happen to control fully. For other sources, living in external code, we should have an options solution for those too. I see two places. In our ActivityListener, or in our ActivityProcessor. I think from an API perspective, something like this would be great:

```csharp
builder.AddActivitySource(HttpWebRequestActivitySource.ActivitySourceName, (options) => 
{
   options.ActivityStarted => (Activity httpActivity) => ...code to modify name...
});
```

We would probably need to change to multiple listeners (instead of one) to make that work cleanly though?